index.html

```diff
@@ -2547,44 +2547,38 @@ <h3>
           be ignored.
         </p>
         <p>
-          The steps for <dfn>processing the <code>purpose</code> member of an
-          image</dfn> are given by the following algorithm. The algorithm takes
-          an <a>ImageResource</a> <var>image</var>. This algorithm will return
-          a <a>set</a> or failure.
+          The steps for <dfn>processing the `purpose` member of an image</dfn>
+          are given by the following algorithm. The algorithm takes an
+          <a>ImageResource</a> |image:ImageResource|. This algorithm returns a
+          [=set=] or failure.
         </p>
         <ol>
-          <li>Let <var>set</var> be an empty <a>set</a>.
+          <li>If [=Type=](|image|["purpose"]) it not String, or
+          |image|["purpose"] consists solely of [=ascii whitespace=], then
+          return the [=set=] « "any" ».
           </li>
-          <li>If <var>image</var> has no <code>"purpose"</code> member, then
-          <a data-lt="set-append">append</a> <code>"any"</code> to
-          <var>set</var>, and return <var>set</var>.
+          <li>Let |keywords:list&lt;string&gt;| be the result of [=split on
+          ASCII whitespace=] |image|["purpose"].
           </li>
-          <li>Let <var>value</var> be <var>image</var>["purpose"].
+          <li>If |keywords| is empty, then return the [=set=] « "any" ».
           </li>
-          <li>Let <a>list</a> <var>keywords</var> be the result of <a>split on
-          ASCII whitespace</a> <var>value</var>.
+          <li>Let |purposes:set| be the empty [=set=].
           </li>
-          <li>If <var>keywords</var> is empty, then <a data-lt=
-          "set-append">append</a> <code>"any"</code> to <var>set</var>.
-          </li>
-          <li>Otherwise, <a>for each</a> <var>keyword</var> of
-          <var>keywords</var>:
+          <li>[=set/For each=] |keyword:string| of |keywords|:
             <ol>
-              <li>Set <var>keyword</var> to <var>keyword</var>,
-              <a>lowercased</a>.
+              <li>Set |canonicalKeyword| to [=lowercased=] |keyword|.
               </li>
-              <li>If <var>keyword</var> is not one of the <a>icon purposes</a>,
-              or <var>set</var> <a>contains</a> <var>keyword</var>, then
-              <a>issue a developer warning</a> and <a>continue</a>.
+              <li>If |canonicalKeyword| is not one of the [=icon purposes=], or
+              |purposes| [=contains=] |keyword|, then [=issue a developer
```

Comment:
FYI: Chrome actually will happily give you back duplicates if we see a valid purpose that is already in the set (we return an array, not a set). It's probably right that the spec doesn't allow duplicates, and we should fix that in Chrome.

Reply:
> Chrome actually will happily give you back duplicates 

When you say "will happily give you back duplicates", you mean internally at the browser level, right?

Reply:
> we return an array, not a set

We do the same... can convert it to list, as it's cleaner.

Reply:
Actually, will leave it... as sets are already lists, so it's an internal implementation detail.

Reply:
Yes, I mean internally in the browser here, sorry for the lack of clarity. :)